pandas/io/json/json.py

```diff
@@ -322,47 +336,143 @@ def read_json(path_or_buf=None, orient=None, typ='frame', dtype=True,
+        self.orient = orient
+        self.typ = typ
+        self.dtype = dtype
+        self.convert_axes = convert_axes
+        self.convert_dates = convert_dates
+        self.keep_default_dates = keep_default_dates
+        self.numpy = numpy
+        self.precise_float = precise_float
+        self.date_unit = date_unit
+        self.encoding = encoding
+        self.lines = lines
+        self.chunksize = chunksize
+        self.nrows_seen = 0
+        self.raw_json = False
+
+        if self.chunksize is not None:
+            self.chunksize = _validate_integer("chunksize", self.chunksize, 1)
+            if not self.lines:
+                raise ValueError("chunksize can only be passed if lines=True")
+
+        if isinstance(filepath_or_buffer, compat.string_types):
+            try:
+                exists = os.path.exists(filepath_or_buffer)
+
+            # gh-5874: if the filepath is too long will raise here
+            except (TypeError, ValueError):
+                exists = False
+
+            if exists:
+                self.data, _ = _get_handle(filepath_or_buffer, 'r',
+                                           encoding=encoding)
+            else:
+                self.raw_json = True
+                self.data = filepath_or_buffer
+        elif hasattr(filepath_or_buffer, 'read'):
+            self.data = filepath_or_buffer
         else:
-            json = filepath_or_buffer
-    elif hasattr(filepath_or_buffer, 'read'):
-        json = filepath_or_buffer.read()
-    else:
-        json = filepath_or_buffer
+            self.raw_json = True
+            self.data = filepath_or_buffer
 
-    if lines:
+    def combine_lines(self, data):
+        """Combines a multi-line JSON document into a single document"""
         # If given a json lines file, we break the string into lines, add
         # commas and put it in a json list to make a valid json object.
-        lines = list(StringIO(json.strip()))
-        json = '[' + ','.join(lines) + ']'
-
-    obj = None
-    if typ == 'frame':
-        obj = FrameParser(json, orient, dtype, convert_axes, convert_dates,
-                          keep_default_dates, numpy, precise_float,
-                          date_unit).parse()
-
-    if typ == 'series' or obj is None:
-        if not isinstance(dtype, bool):
-            dtype = dict(data=dtype)
-        obj = SeriesParser(json, orient, dtype, convert_axes, convert_dates,
-                           keep_default_dates, numpy, precise_float,
-                           date_unit).parse()
-
-    return obj
+        lines = StringIO(data.strip())
+        return '[' + ','.join(lines) + ']'
+
+    def read(self):
+        """Read the whole JSON input into a pandas object"""
+        if self.raw_json:
+            if self.lines:
+                obj = self._get_object_parser(self.combine_lines(self.data))
+            else:
+                obj = self._get_object_parser(self.data)
+        elif self.lines and self.chunksize:
```

Comment:
you need to check lines & chunksize first I think, then determine if its raw_json here